--- src/connect.c       2011-05-31 10:29:52.000000000 -0400
+++ src/connect.c       2012-07-25 23:15:22.281514489 -0400
@@ -159,7 +159,7 @@
   int timeout_ms;
   ssh_pollfd_t fds;
   int rc = 0;
-  unsigned int len = sizeof(rc);
+  socklen_t len = sizeof(rc);

   enter_function();

--- src/misc.c	2012-10-08 23:50:23.567426941 -0400
+++ src/misc.c	2012-10-09 00:19:57.217425822 -0400
@@ -208,17 +208,12 @@
 
 char *ssh_get_user_home_dir(void) {
   char *szPath = NULL;
-  struct passwd pwd;
-  struct passwd *pwdbuf;
-  char buf[NSS_BUFLEN_PASSWD];
-  int rc;
-
-  rc = getpwuid_r(getuid(), &pwd, buf, NSS_BUFLEN_PASSWD, &pwdbuf);
-  if (rc != 0) {
+  struct passwd *pwd = NULL;
+  pwd = getpwuid(getuid());
+  if ( pwd == NULL)
     return NULL;
-  }
 
-  szPath = strdup(pwd.pw_dir);
+  szPath = strdup(pwd->pw_dir);
 
   return szPath;
 }
@@ -233,20 +228,16 @@
 }
 
 char *ssh_get_local_username(ssh_session session) {
-    struct passwd pwd;
-    struct passwd *pwdbuf;
-    char buf[NSS_BUFLEN_PASSWD];
-    char *name;
-    int rc;
-
-    rc = getpwuid_r(getuid(), &pwd, buf, NSS_BUFLEN_PASSWD, &pwdbuf);
-    if (rc != 0) {
+    char *name = NULL;
+    struct passwd *pwd = NULL;
+    pwd = getpwuid(getuid());
+    if ( pwd == NULL) {
         ssh_set_error(session, SSH_FATAL,
             "Couldn't retrieve information for current user!");
         return NULL;
     }
 
-    name = strdup(pwd.pw_name);
+    name = strdup(pwd->pw_name);
 
     if (name == NULL) {
       ssh_set_error_oom(session);
